### DISCUSSION TITLE
Bevy Error (please help)

### DISCUSSION THREAD

**Discussion by astrogames06** [bevyengine/bevy#4921] [Q&A]
## Bevy version

0.7

## What you did

I ran `cargo run`

## Error
```
thread 'main' panicked at 'Failed to initialize any backend! Wayland status: "backend disabled" X11 status: XOpenDisplayFailed', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/linux/mod.rs:619:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## main.rs
``` rust
use bevy::prelude::*;

fn main() {
    App::new()
        .insert_resource(ClearColor(Color::rgb(0.04, 0.04, 0.04)))
        .insert_resource(WindowDescriptor {
            title: "Bevy".to_string(),
            width: 598.0,
            height: 676.0,
            ..Default::default()
        })
        .add_plugins(DefaultPlugins)
        .run();
}
```

# Cargo.toml
``` cargo
[package]
name = "bevy_test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
bevy = "0.7.0"
```

**Comment by bjorn3** [comment]
Looks like you have a Wayland only desktop environment without XWayland support for X11 compatibility. Bevy currently disables Wayland support by default. You can enable it using the `wayland` feature.

```toml
[dependencies]
bevy = { version = "0.7.0", features = ["wayland"] }
```

By the way which distro and desktop environment are you using? I haven't seen any Wayland compositor without XWayland support before.

**Reply by astrogames06** [reply]
I downloaded rust using WSL (Windows Subsystem For Linux) Ubuntu.

**Reply by bjorn3** [reply]
WSL1 or WSL2? WSL1 doesn't support graphics acceleration. It should work for WSL2 out of the box I believe, but it may require a decently up to date windows installation and distro installed in wsl.

**Reply by astrogames06** [reply]
> WSL1 or WSL2? WSL1 doesn't support graphics acceleration. It should work for WSL2 out of the box I believe, but it may require a decently up to date windows installation and distro installed in wsl.

well when i ran ```wsl -l -v``` (this tells you your WSL version) i got this
```
  NAME            STATE           VERSION
* Ubuntu-18.04    Running         2
```
so it looks like its WSL2 but then why isnt bevy working?

**Reply by bjorn3** [reply]
Maybe you could try with the `DISPLAY` env var set to `:0`?

**Reply by astrogames06** [reply]
> Maybe you could try with the `DISPLAY` env var set to `:0`?

i did ```set DISPLAY=0``` and ```set env DISPLAY=0``` but i still get this error
```
thread 'main' panicked at 'Failed to initialize any backend! Wayland status: XdgRuntimeDirNotSet X11 status: XOpenDisplayFailed', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/linux/mod.rs:619:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Reply by bjorn3** [reply]
I meant `:0`, not `0`. The double colon is important.

**Reply by astrogames06** [reply]
> I meant `:0`, not `0`. The double colon is important.

so its ```set DISPLAY=:0``` ?

well because after doing that and running ```cargo run``` i get this
```
ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
ALSA lib conf.c:4528:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4528:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name 
ALSA lib conf.c:4528:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5007:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM default 
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: NoDevice', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_audio-0.5.0/src/audio_output.rs:22:67
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Reply by Pablo1785** [reply]
For me `export DISPLAY=:1` worked (bevy=0.9.0, popos20.04, X11, but the error message looked similar to yours)
